hedera-node/src/main/java/com/hedera/services/ledger/BalanceChange.java

```diff
@@ -344,7 +363,23 @@ public boolean isExemptFromCustomFees() {
         return exemptFromCustomFees;
     }
 
-    public boolean hasNonEmptyAlias() {
-        return accountId.getAccountNum() == 0 && !alias.isEmpty();
+    public boolean hasAlias() {
+        return isAlias(accountId) || hasNonEmptyCounterPartyAlias();
+    }
+
+    public boolean hasNonEmptyCounterPartyAlias() {
+        return isForNft() && counterPartyAccountId != null && isAlias(counterPartyAccountId);
```

Comment:
I think we should remove the `isForNft()` term here since it's true iff `counterPartyAccountId != null`.

Reply:
Done !